quit/web/modules/endpoint.py

```diff
@@ -15,19 +15,42 @@
 
 endpoint = Blueprint('endpoint', __name__)
 
-queryTypeRegex = r"(?P<queryType>("
-queryTypeRegex += r"CONSTRUCT|SELECT|ASK|DESCRIBE|"
-queryTypeRegex += r"INSERT|DELETE|CREATE|CLEAR|DROP|LOAD|COPY|MOVE|ADD"
-queryTypeRegex += r"))"
+queryTypeRegex = r"""
+    (?P<queryType>(CONSTRUCT|SELECT|ASK|DESCRIBE|INSERT|DELETE|
+    CREATE|CLEAR|DROP|LOAD|COPY|MOVE|ADD))
+"""
 queryTypePattern = re.compile(queryTypeRegex, re.VERBOSE | re.IGNORECASE)
 
+mimetypes = {
+    # querytype: { accept type: [content type, serializer_format]}
+    'query': {
+                '*/*': ['application/sparql-results+xml', 'xml'],
+                'application/sparql-results+xml': ['application/sparql-results+xml', 'xml'],
+                'application/xml': ['application/xml', 'xml'],
+                'application/rdf+xml': ['application/rdf+xml', 'xml'],
+                'application/sparql-results+json': ['application/sparql-results+json', 'json'],
+                'text/csv': ['text/csv', 'csv'],
+                'text/html': ['text/html', 'html'],
+                'application/xhtml_xml': ['text/html', 'html']
+    },
+    'construct': {
+                    '*/*': ['text/turtle', 'turtle'],
+                    'text/turtle': ['text/turtle', 'turtle'],
+                    'application/x-turtle': ['application/x-turtle', 'turtle'],
+                    'application/rdf+xml': ['application/rdf+xml', 'xml'],
+                    'application/xml': ['application/xml', 'xml'],
+                    'application/n-triples': ['application/n-triples', 'nt'],
+                    'application/trig': ['application/trig', 'trig']
+    }
+}
+
 
 def parse_query_type(query):
     try:
         query_type = queryTypePattern.search(query).group("queryType").upper()
+        return query_type
```

Comment:
you can make this:
`return queryTypePattern.search(query).group("queryType").upper()`